test/error_handling_test.rb

```diff
@@ -34,4 +34,15 @@ class ErrorHandlingTest < ActiveSupport::TestCase
       ShopifyGraphql.execute(SIMPLE_QUERY)
     end
   end
+
+  test "user error" do
+    fake("mutations/user_error.json", SIMPLE_QUERY)
+
+    begin
+      response = ShopifyGraphql.execute(SIMPLE_QUERY)
+      ShopifyGraphql::Client.new.handle_user_errors(response.data.metafieldDefinitionCreate)
+    rescue ShopifyGraphql::UserError => error
+      assert_equal "TAKEN", error.code
+    end
+  end
```

Comment:
Shouldn't this test fail? Also, shouldn't you rescue `ShopifyAPI::Errors::HttpResponseError` now?

If I rescue `ShopifyGraphql::UserError`, the `error.code` is `200` and not `TAKEN` anymore

To give you more context, I get this error when trying to create a metafield definition that already exist
Before updating the gem to the `2.0.0` version, this code worked great:

```rails
      DEFAULT_METAFIELD_DEFINITIONS.each do |definition|
        CreateMetafieldDefinition.call(**definition)
      rescue ShopifyGraphql::UserError => e
        raise unless e.code == "TAKEN"
      end
```

Now, If I print the `e.code`, I get `200` instead of TAKEN

Reply:
Yeah, this test has been updated in later commit:
https://github.com/kirillplatonov/shopify_graphql/commit/d33d4393000237f5bb5cec30f8ca3998e3dc2b24

And I also added another test for `ShopifyAPI::Errors::HttpResponseError` handling:
https://github.com/kirillplatonov/shopify_graphql/commit/3304bbc074861eeae4d8bad6329a23cf7c6a3f19

In your case, you should switch to new `error_code` method:
```rb
DEFAULT_METAFIELD_DEFINITIONS.each do |definition|
  CreateMetafieldDefinition.call(**definition)
rescue ShopifyGraphql::UserError => e
  raise unless e.error_code == "TAKEN"
end
```

Reply:
Ok, got it. Thank you very much!